addons/website/models/ir_module_module.py

```diff
@@ -518,7 +519,7 @@ def _load_module_terms(self, modules, langs, overwrite=False):
                 specific_arch_db[lang] = field.translate(
                     lambda term: specific_translation_dictionary.get(term, {lang: None})[lang], specific_arch_db_en)
             specific_arch_db['en_US'] = specific_arch_db_en
-            cache.update_raw(View.browse(specific_id), field, [specific_arch_db], dirty=True)
+            cache.update_raw(View.browse(specific_id), field, [{**specific_initial_arch_db, **specific_arch_db}], dirty=True)
```

Comment:
Rather than saving the initial translation values of the specific view to later "repair" the broken dict, I'd prefer to avoid generating that "broken"/"missing some entries" dict, like:
```diff
            langs_update = (langs & generic_arch_db.keys()) - {'en_US'}
+           specific_langs = (langs & specific_arch_db.keys()) - {'en_US'}
            generic_arch_db_en = generic_arch_db.pop('en_US')
            specific_arch_db_en = specific_arch_db.pop('en_US')
            generic_arch_db = {k: generic_arch_db[k] for k in langs_update}
-           specific_arch_db = {k: specific_arch_db.get(k, specific_arch_db_en) for k in langs_update}
+           specific_arch_db = {k: specific_arch_db.get(k, specific_arch_db_en) for k in specific_langs}
```

No?

Reply:
> Rather than saving the initial translation values of the specific view to later "repair" the broken dict, I'd prefer to avoid generating that "broken"/"missing some entries" dict, like:

@rdeodoo Yes, it looks cleaner this way :+1:, code updated!
I also updated the test to handle this use case... Just let me know if something needs to be adjusted :+1: